include/drivers/sensor.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#ifndef DRIVERS_SENSOR_H
+#define DRIVERS_SENSOR_H
+
+#include <dm.h>
+#include <intrusive.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <util.h>
+
+#define SENSOR_OPS(dev) \
+	(&container_of((dev)->drv, struct sensor_driver, drv)->ops)
+
+struct sensor_info {
```

Comment:
This struct serves two purposes:
1. Provide information that will be returned by `SCPI_CMD_GET_SENSOR_INFO`, specifically the "Sensor class" and "Sensor name". For now, since we don't support triggers at all, we always report 0 for "Trigger types", so we don't need to store the "Trigger types" per device.
2. Provide information used by the generic part of the sensor class. For an example from the regulator class, `const struct regulator_range ranges[2];` is stored there, so we don't need to write the code for translation between raw values and cooked values in every driver. It makes sense here to do something similar.

Note that all of the data I've mentioned so far is constant, and is entirely driver-agnostic. There's no cached value (and we have no need to cache values anyway). There's no device-specific address. Anything that's only used by the driver goes in the driver's own struct (like axp803).

Reply:
Readings are no longer stored in the structure. They are now returned by sunxi_thermal_read_raw.